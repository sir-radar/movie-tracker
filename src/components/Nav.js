import React, {useState} from 'react';
import {Link} from 'react-router-dom';

const Nav = () => {

  const [pathname, setPathname] = useState('/');
  console.log(pathname)
  return (
    <nav className="d-flex justify-content-center">
      <Link 
        to='/'
        onClick={ () => setPathname('/')}
        className={ pathname === '/' ? 'p-3 primary-text-color active' : 'p-3 primary-text-color' }
      >Search
      </Link>
      <Link
        to='/favourites'
        onClick={ () => setPathname('/favourites')}
        className={ pathname === '/favourites' ? 'p-3 primary-text-color active' : 'p-3 primary-text-color' }
      >Favorites
      </Link>
      <Link 
        to='/watchlist'
        onClick={ () => setPathname('/watchlist')}
        className={ pathname === '/watchlist' ? 'p-3 primary-text-color active' : 'p-3 primary-text-color' }
      >WatchLater
      </Link>
    </nav>
  );
}

export default Nav;
