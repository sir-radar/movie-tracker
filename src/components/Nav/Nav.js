import React, {useState, useEffect} from 'react';
import {Link} from 'react-router-dom';

const Nav = () => {

  useEffect(() => {
    handlePathChange(window.location.pathname)
  },[]);

  const [pathname, setPathname] = useState('/');

  const handlePathChange = (val) => {
    setPathname(val)
  }

  return (
    <nav className="d-flex justify-content-center">
      <Link 
        data-testid="search"
        to='/'
        onClick={ () => handlePathChange('/')}
        className={ pathname === '/' ? 'p-3 primary-text-color active' : 'p-3 primary-text-color' }
      >Search
      </Link>
      <Link
        data-testid="favourite"
        to='/favourites'
        onClick={ () => handlePathChange('/favourites')}
        className={ pathname === '/favourites' ? 'p-3 primary-text-color active' : 'p-3 primary-text-color' }
      >Favorites
      </Link>
      <Link
        data-testid="watchlater"
        to='/watchlist'
        onClick={ () => handlePathChange('/watchlist')}
        className={ pathname === '/watchlist' ? 'p-3 primary-text-color active' : 'p-3 primary-text-color' }
      >WatchLater
      </Link>
    </nav>
  );
}

export default Nav;
