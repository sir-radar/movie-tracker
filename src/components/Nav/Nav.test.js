import React from 'react';
import { MemoryRouter } from "react-router-dom";
import { render, cleanup, fireEvent } from '@testing-library/react';
import Nav from './Nav';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

afterEach(cleanup)

const renderComponent = () => {
  const div = document.createElement("div");
  return render(<MemoryRouter><Nav/></MemoryRouter>, div)
}

describe('Nav Component', () => {

  it("renders without crashing", () => {
    renderComponent()
  })

  it("renders the router links", () => {
    const { getByTestId } = renderComponent()
    expect(getByTestId('search')).toHaveTextContent("Search")
    expect(getByTestId('favourite')).toHaveTextContent("Favorites")
    expect(getByTestId('watchlater')).toHaveTextContent("WatchLater")
  })

  it("navigates to favourites page when you click the favourite page link", () => {
    const { getByTestId } = renderComponent()
    fireEvent.click(getByTestId('favourite'));
    expect(getByTestId('favourite').classList.contains('active')).toBe(true)
  })

  it("navigates to watchlist page when you click the watchlist page link", () => {
    const { getByTestId } = renderComponent()
    fireEvent.click(getByTestId('watchlater'));
    expect(getByTestId('watchlater').classList.contains('active')).toBe(true)
  })

  it("search page is the first page to render", () => {
    const { getByTestId } = renderComponent()
    expect(getByTestId('search').classList.contains('active')).toBe(true)
  })

  it("matches snapshot", () => {
    const tree = renderer.create(<MemoryRouter><Nav/></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  })

})