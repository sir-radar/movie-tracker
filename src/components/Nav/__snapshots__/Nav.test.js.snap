// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="d-flex justify-content-center"
>
  <a
    className="p-3 primary-text-color active"
    data-testid="search"
    href="/"
    onClick={[Function]}
  >
    Search
  </a>
  <a
    className="p-3 primary-text-color"
    data-testid="favourite"
    href="/favourites"
    onClick={[Function]}
  >
    Favorites
  </a>
  <a
    className="p-3 primary-text-color"
    data-testid="watchlater"
    href="/watchlist"
    onClick={[Function]}
  >
    WatchLater
  </a>
</nav>
`;
